cmd/syft/internal/options/dotnet.go

```diff
@@ -0,0 +1,47 @@
+package options
+
+import (
+	"strings"
+
+	"github.com/anchore/clio"
+	"github.com/anchore/syft/syft/pkg/cataloger/dotnet"
+)
+
+type dotnetConfig struct {
+	SearchLocalLicenses  bool   `yaml:"search-local-licenses" json:"search-local-licenses" mapstructure:"search-local-licenses"`
+	SearchRemoteLicenses bool   `yaml:"search-remote-licenses" json:"search-remote-licenses" mapstructure:"search-remote-licenses"`
+	Providers            string `yaml:"package-providers,omitempty" json:"package-providers,omitempty" mapstructure:"package-providers"`
+	ProviderCredentials  string `yaml:"package-provider-credentials,omitempty" json:"package-provider-credentials,omitempty" mapstructure:"package-provider-credentials"`
```

Comment:
Should this be a structure with fields split out instead of parsing a string? For example, [RegistryCredentials has split out Username and Password](https://github.com/anchore/syft/blob/main/cmd/syft/internal/options/registry.go#L10); we should probably use the same thing for consistency here, and we should use the `secret` type instead of `string` in order to appropriately redact these values everywhere throughout the app. E.g.:

```
type dotnetProviderCredential struct {
	Username secret `yaml:"username" json:"username" mapstructure:"username"`
	Password secret `yaml:"password" json:"password" mapstructure:"password"`
}
type dotnetConfig struct {
	SearchLocalLicenses  bool   `yaml:"search-local-licenses" json:"search-local-licenses" mapstructure:"search-local-licenses"`
	SearchRemoteLicenses bool   `yaml:"search-remote-licenses" json:"search-remote-licenses" mapstructure:"search-remote-licenses"`
	Providers            []string `yaml:"package-providers,omitempty" json:"package-providers,omitempty" mapstructure:"package-providers"`
	ProviderCredentials  []dotnetProviderCredential `yaml:"package-provider-credentials,omitempty" json:"package-provider-credentials,omitempty" mapstructure:"package-provider-credentials"`
}
```

... or maybe the `Providers` makes sense to split into a struct with URL, Username, Password?

Related to the comment about not needing to deal with environment variables later, the `mapstructure` tags in these options is what drives the automatic environment mapping that you will see with `syft config`, however these are a bit funky when it comes to _lists_ such as what I've suggested above. We do, unfortunately end up with some code for that, in a `PostLoad` hook, e.g.: https://github.com/anchore/syft/blob/main/cmd/syft/internal/options/registry.go#L35

Reply:
My initial objective was to make the use of the parameters as easy as possible, hence the orginal decision to use strings, but I do see your point. (Especially the use of the ```secret``` type.) 😉